cfme/middleware/deployment.py

```diff
@@ -0,0 +1,114 @@
+from cfme.fixtures import pytest_selenium as sel
+from cfme.web_ui import listaccordion as list_acc
+from cfme.web_ui.menu import nav, toolbar as tb
+from mgmtsystem.hawkular import Deployment, Path
+from utils.db import cfmedb
+from utils.varmeth import variable
+from . import list_tbl, MiddlewareBase
+
+
+def _db_select_query(name=None, server=None):
+    t_md = cfmedb()['middleware_deployments']
+    query = cfmedb().session.query(t_md.nativeid, t_md.name, t_md.ems_ref)
+    if name:
+        query = query.filter(t_md.name == name)
+    if server:
+        query = query.filter(t_md.nativeid.like('%{}%'.format(server)))
+    return query
+
+
+def _nav_through_provider(ctx):
+    if 'server' in ctx and ctx['server'] is not None:
+        sel.force_navigate('middleware_server_detail',
+                           context={'provider': ctx['provider'], 'name': ctx['server']})
+    else:
+        sel.force_navigate('middleware_provider_detail', context=ctx)
+    list_acc.select('Relationships', 'Middleware Deployments', by_title=False, partial=True)
+
+
+def _nav_to_deployment(ctx):
+    if 'provider' in ctx and ctx['provider'] is not None:
+        _nav_through_provider(ctx)
+    list_tbl().click_row_by_cells({'Deployment Name': ctx['name']})
+
+
+def _select_deployment(ctx):
+    if 'provider' in ctx and ctx['provider'] is not None:
+        _nav_through_provider(ctx)
+    if 'name' in ctx and ctx['name'] is not None:
+        list_tbl().select_row('Deployment Name', ctx['name'])
+
+
+nav.add_branch(
+    'middleware_deployments', {
+        'middleware_deployment': lambda ctx: _select_deployment(ctx),
+        'middleware_deployment_detail': lambda ctx: _nav_to_deployment(ctx)
+    }
+)
```

Comment:
This can be replaced with

```python
from cfme.web_ui.menu import extend_nav

@extend_nav
class middleware_deployments:
    def middleware_deployment(ctx):
        # contents of _select_deployment

    def middleware_deployment_detail(ctx):
        # contents of _nav_to_deployment
```